src/fetch-mock.js

```diff
@@ -91,19 +102,22 @@ function compileRoute (route) {
 			debug('constructing starts with string matcher for route: ' + route.name);
 			expectedUrl = expectedUrl.substr(1);
 			route.matcher = function (url, options) {
-				return matchMethod(options) && url.indexOf(expectedUrl) === 0;
+				var req = toRequest(url, options);
+				return matchMethod(req) && req.url.indexOf(expectedUrl) === 0;
```

Comment:
could you change the parameter name in matchMethod's definition to be `req` too please